### DISCUSSION TITLE
Auto Dark Mode 10.4.0

### DISCUSSION THREAD

**Discussion by Spiritreader** [AutoDarkMode/Windows-Auto-Night-Mode#767] [Announcements]
This update introduces a full UI refresh that closes the gap between ADM and Windows 11.
You can also use Auto Dark Mode to set your system accent colors and mouse cursor!
In addition, we now fully support Windows spotlight wallpapers on Windows 11 22621.1105 and newer.
There are also many QoL improvements and of course, bug fixes!

## Important Updater Notice
With 10.4, we've improved the update system and made it more future proof. Because this introduced incompatibilities with the old system, you can't use the automatic updater if you are on version 10.3. In this case, just download the installer below and re-install Auto Dark Mode.

## Known limitations
Auto Dark Mode comes with a few limitations, either due to Windows or because of technical reasons. 
You can find the full list here: [List of known limitations](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/wiki/Known-limitations)

# Changelog (10.4.0.35)

## New Features
- Full spotlight support in Windows builds `>=22621.1105` for classic and theme mode!
- Support for setting system accent colors!
- Support for setting mouse cursors!
- Rudimentary UI for turning scripts on and off, as well as opening the configuration file.
- The ADM tray will now show access keys when opened (by @Susko3 #664).
- On Windows 11 22H2 and up, touch keyboard light/dark theme switching can be used again by enabling the toggle on the Apps page (by @Armin2208).
- ADM can now listen for running processes and postpone automatic theme switches (by @pohuing #733 #734).

## Improvements
- The UI now follows the Win11 design language more closely, all pages have been updated.
- UI elements will scale out properly when maximizing the window.
- The wallpaper page will now preselect the light/dark combobox based on the currently active theme.
- ADM will attempt to auto-repair invalid/corrupt windows theme files.
- Auto skip postpone entries are no longer created on a theme switch toggle if a user delay is present.
- Increased the robust-ness of theme file parsing and added auto-repair mechanisms!
- The wallpaper monitor selector will now show which displays are currently not connected!
- You can now choose between Fill, Fit and Stretch in multi monitor wallpaper picker.
- Postpones will be preserved after closing Auto Dark Mode.
- Auto Dark Mode is now able to react to timezone changes while it is running.
- The update process has been streamlined.
- Display Detection has been improved to reduce occurrences when to many monitors were detected during a switch.

## Fixes
- An issue was fixed that caused the theme file apply mitigation patch to increase values to a point where users would notice.
- Fixed an issue where wallpapers in theme files weren't parsed correctly under certain circumstancese.
- Downgrading from the beta branch will now correctly show a notification even when silent updates are enabled.
- The shell will no longer crash when resizing the window when the new Windows terminal app is used.
- Timestamps will now use the correct UI culture.
- An issue was fixed that caused metadata in unmanaged themes to be set incorrectly.
- DWM will now refresh when the "Accent color for titlebars and window colors" setting is modified in the UI.
- Pressing the copy version information will now attempt to retry accessing the clipboard if it is unavailable.
- The shell timeout parameter will now be correctly retrieved when there are multiple arguments
- System resume events will no longer trigger an automatic theme switch when automation is disabled.
- A critical issue was fixed that caused offsets to be applied to custom times.
- An issue was fixed that caused the named pipe server to not execute commands if the client didn't wait for the response pipe.
- Management Event Watchers are now disposed of correctly in the UI.
- Fixed an issue where themes would not refresh during dark mode if auto switch has been paused.
- Fixed an issue that caused internal config updates to still perform a save operation.
- Fixed an issue that caused native titlebars to shine through when accent color for titlebars and window borders was enabled.
- Fixed an issue that caused the GPU Monitoring module to activate at the wrong times when night light is selected.
- Fixed an issue where the timepicker was disabled after re-enabling automatic theme switch (by @Armin2208).
- Fixed an issue that caused ADM to remain open when an exit instruction was sent via shell before all internal components were fully initialized.

## Changes
- The battery detection mechanism has once again been changed, it will now use the power supply status as its source.
- The wording for the UI setting relatd to  "Accent color for title bars and window borders" was changed.
- Debug logs will be enabled when the beta branch is selected.
- The internal component structure was revamped, all theme switch arguments are now consolidated in a single EventArg class.
- The BaseComponent class has been extended and now includes a callback method.
- The App will now force the same locale as the service on every startup. This should prevent desynchronizations
- Language for the App now only changes when the restart button is pressed for consistency with the service behavior.
- GPU monitoring postpones can now be cleared from the ADM app if desired.
- The color filter activation setting has been moved to the Apps page.

__________

**Full Changelog**: https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/compare/10.3.0.90...10.4.0.35

## Translation Updates
Translations and translation changelogs have been moved to weblate, thanks again to all of our translator volunteers!
https://hosted.weblate.org/projects/auto-dark-mode/app/

<details>
  <summary>Translation history</summary>

Disclaimer: May not be complete as weblate does not allow for automatic changelog generation in a readable privacy-protected format.

- Turkish
    - Oğuz Ersen (1)
    - cumhur (68)

- Portuguese (Brazil)
    - brunocardosoeng91 (7)
    - Mayara da Silva Oliveira (25)

- German
    - Felix Koch (1)
    - Sebastian Espei (14)
    - Sam (724)

- Greek
    - Λευτέρης Τ. (2)
    - Thunderstrike116 (59)

- English
    - Sam (52)

- Russian
    - Alloyd (4)
    - Алексей Рыбников (4)
    - lenemter (31)

- Romanian
    - Emanuel (5)

- Ukrainian
    - Dan (65)

- Japanese
    - maboroshin (104)

- Vietnamese
    - tictactoe (1)
    - ngocanhtve (1)
    - Khoi Le (6)

- Hungarian
    - Botond Csatári-Szűcs (183)

- Hebrew
    - moshe naamad (220)

- Polish
    - Tomasz Małota (1)
    - Eryk Michalak (7)

- Italian
    - Giacomo Farini (1)
    - J. Lavoie (9)

- Indonesian
    - Reza Almanda (2)
    - pc-v2 (102)

- Dutch
    - DDG (10)
    - jerone (33)
    - QkeleQ10 (73)

- Spanish
    - Daniel (3)
    - gallegonovato (52)

- French
    - Fyihang (1)
    - CedriCDAR77 (2)
    - J. Lavoie (5)
    - Damien Trouillet (12)
    - Maxence C. (13)
    - Pipixel_06 (56)

- Serbian
    - bzzrak (42)
    - Dragan (398)

- Korean
    - Min Ki Jung (Ganke) (59)

- Portuguese (Portugal)
    - Duarte Flores (2)
    - tluciomiranda (49)

- Chinese (Traditional)
    - Kane (1)
    - stone wow (2)
    - Fyihang (3)
    - jrthsr700tmax (4)
    - hugoalh (178)

- Chinese (Simplified)
    - 星河 (1)
    - Jus Orange (OrangeJus) (4)
    - anye1998 (6)
    - JY3 (8)
    - Eric (21)
    - ThrRip (34)

</details>

<hr /><em>This discussion was created from the release <a href='https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/releases/tag/10.4.0.35'>Auto Dark Mode 10.4.0</a>.</em>

**Comment by gherbetto** [comment]
Amazing work, been rocking this since it came out, since Microsoft won't make it a built in feature. Apple made it happen in Catalina in 2019. Cheers!

**Comment by PeopleInside** [comment]
Why the software alert me about the update with a notification and redirect me to the download page?
In settings I have selected to auto update the program itself and i want avoid to get the update notification.

**Reply by Spiritreader** [reply]
It's in the release notes. I haven't updated the discussion page description yet tho,

**Comment by beancaper** [comment]
Will you be updating the Microsoft Store version?

**Reply by Armin2208** [reply]
It's updated, but Microsoft Store doesn't support updating classic Apps. So you need to re-download Auto Dark Mode.

**Comment by phrichuk** [comment]
Great work! When I install 10.4.0.35, winget update and Control Panel Uninstall dialogue still say I have 10.4.0.0. It works fine, but it confuses winget! **Is there a mismatch somewhere in the latest installer that has retained the earlier version number?**
![image](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/assets/70862657/65b8bd07-2a2c-4894-ba5e-94e64782cfaf)
![image](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/assets/70862657/0fbfdb45-6d35-4c34-8e31-61fd63c05b9b)
![image](https://github.com/AutoDarkMode/Windows-Auto-Night-Mode/assets/70862657/d8f7f868-489d-4f19-82f3-cb93e82aaa84)

**Reply by Spiritreader** [reply]
No that's all correct. There are some issues with the inno setup installer.

The actual correct build number is baked into the executable's assembly, which is 10.4.0.35.

The winget package has 10.4.0.35 set as the version number, which doesn't match the inno setup 10.4.0.0 version string.
The package is community maintained, so whoever is maintaining it needs to change the reported version to that which matches the inno setup string.

We'll probably get inno setup to pick up the correct version at some point, but replacing all executables means that all package installers need to update the hash.

The best you can do atm is pinning the adm package with winget, for more info check out this issue: #771
